### DISCUSSION TITLE
[Bug]: inaccurate information drawn into "Location" panel of WP meetings details page.

### DISCUSSION THREAD

**Discussion by WMikeCarroll** [code4recovery/12-step-meeting-list#535] [General]
### Contact Details

sc-events@comcast.net

### Website With Issue

ContraCostaAA.org

### What happened?

For Zoom Meetings only - 
![ElSobProbs](https://user-images.githubusercontent.com/90402795/134823489-d944b030-1ede-4d19-bad4-8ed9709c0a46.JPG)
)
Inaccurate, apparently info is drawn into the "locations" panel on the meetings detail page
[Meetings CoCoCounty 9-26-2021.xls](https://github.com/code4recovery/12-step-meeting-list/files/7232218/Meetings.CoCoCounty.9-26-2021.xls)
I have attached our Excel file for anyone interested.

### TSML Version

Latest (default)

### Wordpress Version

Latest (default)

### What browsers are you seeing the problem on?

Chrome

### Relevant log output/errors

```shell
None
```

**Comment by Grateful-Viv** [comment]
Shouldn't address contain the street address, city, state, and zip code? I just downloaded our meetings and there is no city field. Maybe the template is different and the city, state, zip code fields are different.

But if this is an online meeting I am curious why you have 94803 in the zip code field and CA in the State field for the Monday El Sobrante Fellowship 830AM meeting.

**Reply by tech2serve** [reply]
The city field is optional in the spec *if* the entire address is entered into the address field. This is the preferred way for TSML. This is why you don't see it in the exported csv.

**Comment by tech2serve** [comment]
@WMikeCarroll,

For an online meeting you *must* still put in the city and state (as a non-specific address) along with the online meeting details.

If the meeting can also be attended in person, you need to enter the specific physical address.

Are you importing meeting information from a spreadsheet?

===Edit===
Looking at the spreadsheet, it appears to be the cause of your issues.

There are two sources of information for you to correct the problems. The first is the example spreadsheet provided by TSML in the Import & Settings area. The second is the [spec for the Meeting Guide](https://github.com/code4recovery/spec), to which TSML adheres.

If after you've made the corrections, and you're still having troubles, please let us know.

**Reply by Grateful-Viv** [reply]
I thought you could put the county and state for a natively online meeting. If the meeting has the location closed, but is meeting online, then enter address information, then you at least enter the meeting URL or the phone number , but mark off that the meeting is meeting not in-person. Not sure how to enter this in the spreadsheet to import rather than just doing in from WordPress. But I assume if the fields are entered properly it should know.

**Reply by tech2serve** [reply]
I've never tested just using county and state. Unless I've goofed, every time I've described these requirements (and in the edit meeting page) the admin should enter city and state for online meetings.

If the meeting cannot be attended in person, the admin should not enter a specific address (unless TC of course).

In the case for this issue though, looking at the spreadsheet of data shows headings mislabeled and the Address only has City information. Without having a state include (i.e., "Bakersfield, CA"), I suspect the Google geocoder doesn't know how to handle this, and the results are unpredictable.

**Reply by Grateful-Viv** [reply]
> I've never tested just using county and state. Unless I've goofed, every time I've described these requirements (and in the edit meeting page) the admin should enter city and state for online meetings.

Back when we were beta testing, and ever since on the live versions, all I’ve ever entered was county and state. Works like a charm.

**Reply by joshreisner** [reply]
please also see https://github.com/code4recovery/12-step-meeting-list/discussions/521

**Reply by tech2serve** [reply]
Ha! Thank you. I *thought* I had answered this previously!

**Comment by WMikeCarroll** [comment]
Dear Code4Recovery,

Thanks for your help.

In my defense, when I first had the problem with extraneous information 
being pulled into the Location field on the WP pages, I was told to put 
the City in the Location field.  Recently I learned that I also needed 
the state and zip in that field, which I have done, to some benefit.

I also read the term  "geocode" for the first time.

I am learning, but apparently not enough.

Here are two pages assembled by WP from our database showing incorrect 
information.

For the El Sobrante Fellowship (which is not in Richmond):

For Peanut Butter and Jelly:


This is a snip from the Excel file that gets converted to a CSV before 
uploading.Note that "Richmond" and “Education room @L" is not in either 
of these.

These are zoom meetings (URL not included in this snip).

You will see that I have added the state and zip to the City name in the 
Address field, as instructed.

So how do I keep WP from pulling extraneous info into these (and many 
other) pages?

I look forward to any guidance you can give.

All the best, and thanks again,
Mike C
Contra Costa Service Center Web Lackey

**Reply by tech2serve** [reply]
It's all good Mike. I know learning can be tricky; I'm banging my head on the wall daily over some new concepts I'm trying to master. :-)

Are you still populating `Location` with "May be inaccurate"? If so, try removing this and see if any of the problems clear up. I'm a little concerned the internal cache may be contributing now.

Also, please make sure the meetings at El Sobrante Fellowship that are available in-person have the specific street address, city and state in the Address field.

If once you've confirmed those changes, you're still having extraneous data show on the webpage, please provide another copy of the spreadsheet and indicate for me which row (i.e., meeting) is having the problem.

**Comment by Grateful-Viv** [comment]
Is there a particular reason why you are keeping your meeting info current by importing a spreadsheet rather than just going through WordPress directly? 

I did a couple of test meetings directly in the WordPress back-end. 

In one example, I entered Sobrante, CA in the address field and the other Contra Costa County, CA in the address field. Both came out as expected. See below screenshot.
![El-Sobrante-Fellowship-–-Southern-Delaware-Intergroup-Staging-Site](https://user-images.githubusercontent.com/62247421/135463140-878e34e7-f206-400d-b179-b35154d6956b.png)

I am also curious what category this online meeting falls under. There are 3 types

(1) Used to meet at a physical location, but currently temporarily closed
(2) Meeting is both online and in-person at a physical location (hybrid)
(3) Meetings without a street address but an approximate location (natively-online)

If this particular meeting falls under number 3, then all you need is the conference URL (Zoom link) and an approximate city & state (or I still believe county & state works). If the location name is the same for multiple meetings, then you will see displayed xx number of meetings at this location (which is another issue for another time - see #371). Where it asks "Can I currently attend this meeting in person?", you would check no.

If this meeting falls under number 1, then you would enter the conference URL (Zoom link) and all the geocodable info about the location, including actual street address, but like above, where it asks "Can I currently attend this meeting in person?", you would check no. Then you would see displayed the address with a line crossed out.

I am not one of the Code for Recovery people, just an end user trying to be helpful. I am sure the developers will chime in if something here is incorrect.

**Reply by tech2serve** [reply]
I have been vocal that I am not a fan of recurring imports of spreadsheets into TSML :-) I know there are some users who adamantly disagree with me, and they often have valid reasons.

But users who choose this route often seem to have more difficulties. Why? Because understanding how to correctly map fields *is* more challenging than simply learning the user interface. 

That said, Mike may be just trying to get the initial load of data working in TSML, and this is a viable use case for importing a csv.
